data/src/main/java/io/quarkus/qe/data/RepositoryEntity.java

```diff
@@ -36,4 +61,13 @@
     @ManyToOne
     @JoinColumn(name = "quarkus_version_id")
     public QuarkusVersionEntity quarkusVersion;
+
+    public static final Predicate<RepositoryEntity> byAnyExtensionWithNameAndVersionIfNotEmpty(String name, String version) {
+        return repository -> repository.extensions.stream()
+                .anyMatch(QuarkusExtensionEntity.byNameAndVersionIfNotEmpty(name, version));
+    }
+
+    public static final Predicate<RepositoryEntity> byAnyLabel(String label) {
```

Comment:
Same as above.

Reply:
I do want to prevent/avoid a subclass to redefine this method. 
Also, as this is an entity, without static and final, I would need to mark the method as `@Transient`.